lib/multi.c

```diff
@@ -1223,6 +1306,33 @@ static CURLMcode Curl_multi_wait(struct Curl_multi *multi,
         extra_fds[i].revents = mask;
       }
 
+#ifdef USE_WINSOCK
+      /* Count up all our own sockets that had activity,
+         and remove them from the event. */
+      if(curlfds) {
+        data = multi->easyp;
+        while(data) {
+          bitmap = multi_getsock(data, sockbunch);
+
+          for(i = 0; i < MAX_SOCKSPEREASYHANDLE; i++) {
+            if(bitmap & (GETSOCK_READSOCK(i) | GETSOCK_WRITESOCK(i))) {
+              wsa_events.lNetworkEvents = 0;
+              if(WSAEnumNetworkEvents(sockbunch[i], NULL, &wsa_events) == 0) {
+                if(ret && pollrc <= 0 && wsa_events.lNetworkEvents != 0)
+                  retcode++;
+              }
+              WSAEventSelect(sockbunch[i], multi->wsa_event, 0);
+            }
+            else
+              break;
```

Comment:
Why are you breaking out of this, isn't multi poll supposed to return the total number of sockets on which events have happened, and wouldn't that require checking all the sockets?

Reply:
Yes, but the bitmap tells how many sockets we were supposed to check in the first place. So this is just the same condition as for the loops in the setup of the waiting at the beginning of the function.

Reply:
Wait a moment, you are right. This is not exactly the same condition as in the setup of the function. Also see my comment to @bagder above that there is another condition in the setup.

Reply:
Resolved, see https://github.com/curl/curl/pull/6245#issuecomment-797323267.